ludicrousdb/includes/class-ludicrousdb.php

```diff
@@ -1151,16 +1141,28 @@ public function flush() {
 	public function check_connection( $allow_bail = true, $dbh_or_table = false, $query = '' ) {
 		$dbh = $this->get_db_object( $dbh_or_table );
 
-		if ( $this->dbh_type_check( $dbh ) ) {
-			if ( true === $this->use_mysqli ) {
-				if ( mysqli_ping( $dbh ) ) {
-					return true;
-				}
-			} else {
-				if ( mysql_ping( $dbh ) ) {
-					return true;
+		// we only want to check the connection once per page load, to avoid mysqli_ping slowing down things
+		// this means if the read/write server goes down while the page is loaded, we will have an mysql error instead of graceful failover
+		// however this will happen so rarely & isn't an issue (bc user can just refresh the page) that it's not worth to test/mysqli_ping for every read/write
+		$check_connection = wp_cache_get( $dbh->thread_id, 'non_persistent' );
+		// if the connection was not checked or not established, try to check the connection
+		if( $check_connection === false ) {
+			if ( $this->dbh_type_check( $dbh ) ) {
+				if ( true === $this->use_mysqli ) {
```

Comment:
I would suggest keeping your conditionals the same, either use yoda conditions or don't.

Reply:
It happened bc I copied the part from WP core, which used to use yoda (and not all locations are changed yet)
Thanks though